fix(sidebar): Fetch org details in user settings

maheskett opened the pull request:
- During the require 2FA flow, if the user doesn't have an organization and needs to setup 2FA, display the user name and email at the top of the sidebar dropdown.
- In account settings, organization can be pulled from the index endpoint instead of details. Fetch organization details so the sidebar dropdown has `access` information and displays `Members`, `Teams`, etc. tabs.

billyvg commented on src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityEnroll.jsx:
```diff
@@ -165,8 +165,8 @@ class AccountSecurityEnroll extends AsyncView {
       fetchOrganizationByMember(this.invite.memberId, {
-        loadOrg: true,
-        setActive: true,
+        addOrg: true,
```
Is this the only place we call `fetchOrganizationByMember`?

maheskett replied:
Yes

billyvg commented on src/sentry/static/sentry/app/views/settings/account/accountSettingsLayout.jsx:
```diff
@@ -1,9 +1,31 @@
+    // if there is no org in context, SidebarDropdown uses an org from `withLatestContext`
+    // (which queries the org index endpoint instead of org details)
+    // and does not have `access` info
+    if (organization && !organization.access) {
```
I think `access` is always an array, but we could be a bit more clear and check that `access` is explicitly not defined on `organization` instead of a general falsey value.

billyvg commented on tests/js/spec/views/settings/accountSettingsLayout.spec.jsx:
```diff
@@ -0,0 +1,46 @@
+  let spy;
+  let api;
+
+  let organization = {
```
Are we defining this org to better reflect the "summary" org (vs org details)?

maheskett replied:
Ya that's what I was going for.

billyvg commented:
Can we add a PR description?

billyvg commented:
`api tokens - no tokens` is also interesting....

maheskett commented:
Updated the description. The `api tokens - no tokens` diff is intentional, but not sure what's going on with the onboarding bar in percy. It's there locally so maybe it's loading after the snapshot...

maheskett commented:
Fixed the percy tests. For posterity, the issue was around which features the tests had access to.

Also, saw that `/settings/` didn't have org details so I added a call there. What do you think about merging this now and then I can followup to fetch org details higher up, maybe in `SettingsWrapper`?

billyvg commented:
@maheskett I believe @ckj wants to remove that page
